### PR TITLE
Add CGAN

### PR DESCRIPTION
**Pull request by saksham189**

### CODE REVIEW

**Comment by saksham189** [comment]
@ShikharJ  currently for generating samples from GAN we do `gan.Generator().Forward(noise, samples);`. However for cgan we also need to pass the labels and concatenate them. Do you think we should add a method `Forward(noise, labels, samples)` specific to CGAN for generating samples.

**Comment by zoq** [comment]
> @ShikharJ currently for generating samples from GAN we do `gan.Generator().Forward(noise, samples);`. However for cgan we also need to pass the labels and concatenate them. Do you think we should add a method `Forward(noise, labels, samples)` specific to CGAN for generating samples.

Personally I like the idea, makes is easier to generate samples.

**Comment by saksham189** [comment]
@ShikharJ this is ready for review.

File: src/mlpack/methods/ann/gan/gan_policies.hpp [hunk 1]
```diff
@@ -78,6 +78,10 @@ class WGAN { /* Nothing to do here */ };
  */
 class WGANGP { /* Nothing to do here */ };
 
+
+
+class CGAN { /* Nothing to do here */ };
```
Comment (ShikharJ) [inline src/mlpack/methods/ann/gan/gan_policies.hpp]: Can this be put in alphabetic order?

**Comment by saksham189** [comment]
> Thanks for the contribution, I'd also appreciate if you could provide the output of the GAN as well to check if everything works well.

Hey, @ShikharJ thanks for the review. I think that Toshal has a PR open regarding fixing the gradient of the GAN so, we should test that first and then proceed to testing this. 
Also I think that the training would take many days if I do it locally on my machine. Would you be able to provide me access to the server that Toshal was using for testing this.

**Comment by rcurtin** [comment]
This code touches the ANN code, and there was recently a big refactoring of the ANN code in #2259, so be sure to merge the master branch into your branch here to make sure that nothing will fail if this PR is merged. :+1: (I'm pasting this message into all possibly relevant PRs.)

**Pull request closed without merging**